test/distributed/test_c10d_functional_native.py

```diff
@@ -326,23 +330,44 @@ def test_all_to_all_single(self) -> None:
         assert output.eq(expect).all()
         assert output.completed
 
+
+class C10DFunctionalNativeCompileTest(TestCase):
+    def setUp(self):
+        # Allow testing aoti after torch.compile
+        torch._inductor.config.triton.store_cubin = True
+        torch._inductor.config.debug = True
+
+        self.rank = 0
+        self.world_size = 2
+        torch.cuda.set_device("cuda:0")
+
+        store = FakeStore()
+        dist.init_process_group(
+            backend="fake",
+            world_size=self.world_size,
+            rank=self.rank,
+            store=store,
+        )
+        funcol.enable_native_funcol()
+
+    def tearDown(self):
+        funcol.disable_native_funcol()
+        dist.destroy_process_group()
+
     @unittest.skipIf(not has_triton(), "Inductor+gpu needs triton and recent GPU arch")
-    @torch._inductor.config.patch(debug=True)
     @fresh_inductor_cache()
     def test_inductor_all_reduce_single(self):
-        self._init_process_group()
-
         def func(arg: torch.Tensor) -> torch.Tensor:
             buf0 = arg + 42
             # Expect in-place with inductor allocated buf
-            ar0 = funcol.all_reduce(buf0, "avg", "default")
+            ar0 = funcol.all_reduce(buf0, "avg", "0")
```

Comment:
if you use a fakePG as your only pg does "default" no longer map to it?

Reply:
Preivously, this test explicitly registered the global pg under the name "default" (this was before auto name registration was implemented so I had to bootstrap). By default, the global pg gets the name "0" (this something we could discuss and change).